grammars/javascript.cson

```diff
@@ -281,240 +281,283 @@
         'name': 'variable.language.js'
   }
   {
-    'captures':
-      '1':
-        'name': 'support.class.js'
-      '2':
-        'name': 'support.constant.js'
-      '3':
-        'name': 'keyword.operator.assignment.js'
-    'comment': 'match stuff like: Sound.prototype = { … } when extending an object'
-    'match': '([a-zA-Z_?.$][\\w?.$]*)\\.(prototype)\\s*(=)\\s*'
-    'name': 'meta.class.js'
-  }
-  {
-    'begin': '([a-zA-Z_?.$][\\w?.$]*)\\.(prototype)\\.([a-zA-Z_?.$][\\w?.$]*)\\s*(=)\\s*(?:(async)(?:\\s+))?(function\\*?)\\s*(\\*?)(\\()'
-    'beginCaptures':
-      '1':
-        'name': 'support.class.js'
-      '2':
-        'name': 'support.constant.js'
-      '3':
-        'name': 'entity.name.function.js'
-      '4':
-        'name': 'keyword.operator.assignment.js'
-      '5':
-        'name': 'storage.modifier.js'
-      '6':
-        'name': 'storage.type.function.js'
-      '7':
-        'name': 'storage.type.function.js'
-      '8':
-        'name': 'punctuation.definition.parameters.begin.js'
-    'comment': 'match stuff like: Sound.prototype.play = function() { … }'
-    'end': '(\\))'
-    'endCaptures':
-      '1':
-        'name': 'punctuation.definition.parameters.end.js'
-    'name': 'meta.function.prototype.js'
+    # [async] function [name](params)
+    # function* name(params) – generator function declaration
+    'begin': '(?=(?:\\basync\\b\\s*)?\\bfunction\\b)'
```

Comment:
No need for `?:` non-capturing group because we're not capturing anything here anyway.